ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -2960,7 +2982,8 @@ static vk_device ggml_vk_get_device(size_t idx) {
 
         vkGetPhysicalDeviceFeatures2(device->physical_device, &device_features2);
 
-        device->fp16 = device->fp16 && vk12_features.shaderFloat16;
+        // GCN 3 and 4 chips support FP16 at regular speed, but the drivers don't indicate it
+        device->fp16 = device->fp16 && (vk12_features.shaderFloat16 || (device->architecture == AMD_GCN34));
```

Comment:
It's against the spec to enable this if support is not indicated with the `shaderFloat16` feature, which it is not for these GPUs. Did you try running this with validation layers enabled? I'm relatively sure this should throw validation issues.